internal/caravan/tradeagreement.lua

```diff
@@ -26,13 +86,71 @@ local function diplomacy_toggle_cat()
     end
 end
 
+local function OrderGemWithMinValue(matPrices, val)
+    local priority_idx = GetCurrentTabId()
+
+    local priority = diplomacy.environment.dipev.sell_requests.priority[priority_idx]
+    for i in ipairs(priority) do
+        if matPrices[i] == val then
+            priority[i] = 4
+        end
+    end
+end
+
+local function GetCivMatPrices(tabSelected)
+    local resource = tabSelected.getCivResourceList(df.historical_entity.find(diplomacy.actor.civ_id).resources)
+    local matPrices = {}
+    local matValuesUnique = {}
+    local filter = {}
+    for civid, matid in pairs(resource) do
+        local matPrice = tabSelected.funcGetValue(matid)
+        matPrices[civid] = matPrice
+        if not filter[matPrice] then
+            local val = { value = matPrice, count = 1 }
+            filter[matPrice] = val
+            table.insert(matValuesUnique, val)
+        else
+            filter[matPrice].count = filter[matPrice].count + 1
+        end
+    end
+    table.sort(matValuesUnique, function(a, b) return a.value < b.value end)
+    return matPrices, matValuesUnique
+end
+
+local function ValueSelector()
+    local currTab = GetTab()
+    local matPrices, matValuesUnique = GetCivMatPrices(currTab)
+    local list = {}
+    for index, value in ipairs(matValuesUnique) do
+        list[index] = tostring(value.value) ..
+            " - " .. tostring(value.count) .. " " .. (value.count == 1 and currTab.labelSingular or currTab.labelPlural)
```

Comment:
for multi-part strings like this, it is usually cleaner to use a format string:
```lua
('%d - %d %s'):format(value.value, value.count,
    value.count == 1 and currTab.labelSingular or currTab.labelPlural)
```

Reply:
I'd also suggest adding the "gear" symbol after the value to indicate price. the number-heavy string label also feels a bit user-unfriendly. we might want to post some more screenshots to the #feature-discuss Discord channel to get feedback on that.